### PR TITLE
G-HIP-39 | Adding v1/consents/hip/notify API

### PR DESCRIPTION
**Pull request by shwetadongare**

### CODE REVIEW

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/Consent/ConsentArtefactRepresentation.cs  1
- src/In.ProjectEKA.HipService/Consent/Model/Notification.cs  1
- src/In.ProjectEKA.HipService/Consent/ConsentNotificationController.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5600159&bid=18347645)

**Comment by ganesanarun** [comment]
Tests? :)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/Consent/ConsentArtefactRepresentation.cs  1
- src/In.ProjectEKA.HipService/Consent/Model/Notification.cs  1
- src/In.ProjectEKA.HipService/Consent/ConsentNotificationController.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5600159&bid=18347645)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/Consent/ConsentArtefactRepresentation.cs  1
- test/In.ProjectEKA.HipServiceTest/Consent/ConsentNotificationControllerTest.cs  2
- src/In.ProjectEKA.HipService/Consent/ConsentNotificationController.cs  2
- src/In.ProjectEKA.HipService/Consent/Model/Notification.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5600159&bid=18347645)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/Consent/ConsentNotificationController.cs  2
- test/In.ProjectEKA.HipServiceTest/Consent/ConsentNotificationControllerTest.cs  2
- src/In.ProjectEKA.HipService/Consent/ConsentArtefactRepresentation.cs  1
- src/In.ProjectEKA.HipService/Consent/Model/Notification.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5600159&bid=18347645)

File: test/In.ProjectEKA.HipServiceTest/Consent/ConsentNotificationControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        public ConsentNotificationControllerTest()
+        {
+            consentRepository = new Mock<IConsentRepository>();
+            backgroundJobClient = new Mock<IBackgroundJobClient>();
+            consentNotificationController = new ConsentNotificationController(consentRepository.Object,
+                                                                              backgroundJobClient.Object);
+        }
+        
+        [Fact]
+        private async void ShouldEnqueueLinkRequestAndReturnAccepted()
```
Comment (ganesanarun) [inline test/In.ProjectEKA.HipServiceTest/Consent/ConsentNotificationControllerTest.cs:36]: why do we need async here?

Comment (Nexengineer) [reply test/In.ProjectEKA.HipServiceTest/Consent/ConsentNotificationControllerTest.cs:36]: Not need Removed

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- src/In.ProjectEKA.HipService/Consent/ConsentArtefactRepresentation.cs  1
- src/In.ProjectEKA.HipService/Consent/ConsentNotificationController.cs  2
- src/In.ProjectEKA.HipService/Consent/Model/Notification.cs  1
- test/In.ProjectEKA.HipServiceTest/Consent/ConsentNotificationControllerTest.cs  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5600159&bid=18347645)

**Comment by ganesanarun** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- test/In.ProjectEKA.HipServiceTest/Consent/ConsentNotificationControllerTest.cs  2
- src/In.ProjectEKA.HipService/Consent/ConsentNotificationController.cs  2
- src/In.ProjectEKA.HipService/Consent/ConsentArtefactRepresentation.cs  1
- src/In.ProjectEKA.HipService/Consent/Model/Notification.cs  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/ProjectEKA/hip-service/pullRequest?prid=5600159&bid=18347645)

**Pull request merged**